persistence/mongodb/src/main/scala/net/liftweb/mongodb/MongoAsync.scala

```diff
@@ -0,0 +1,91 @@
+package net.liftweb.mongodb
+
+import java.util.concurrent.ConcurrentHashMap
+
+import com.mongodb.MongoException
+import com.mongodb.async.client.{MongoClient, MongoCollection, MongoDatabase}
+import com.mongodb.async.SingleResultCallback
+import net.liftweb.util.ConnectionIdentifier
+import org.bson.Document
+import org.bson.codecs.{IntegerCodec, LongCodec, ObjectIdCodec, StringCodec}
+import org.bson.codecs.configuration.CodecRegistries
+
+import scala.concurrent.Promise
+
+private[mongodb] class SingleBooleanVoidCallback(f: () => Unit) extends SingleResultCallback[Void] {
+  private[this] val p = Promise[Boolean]()
+  override def onResult(result: java.lang.Void, t: Throwable): Unit = {
+    if (t == null) {
+      f()
+      p.success(true)
+    }
+    else p.failure(t)
+  }
+  def future = p.future
+}
+/*
+  Async version of MongoClient
+ */
+object MongoAsync {
+
+  /*
+  * HashMap of Mongo instance and db name tuples, keyed by ConnectionIdentifier
+  */
+  private[this] val dbs = new ConcurrentHashMap[ConnectionIdentifier, (MongoClient, String)]
+  val codecRegistry = CodecRegistries.fromRegistries(com.mongodb.MongoClient.getDefaultCodecRegistry(),
+    CodecRegistries.fromCodecs(new LongPrimitiveCodec, new IntegerPrimitiveCodec)
```

Comment:
without that codecs old-style (2.x with BasicDbObject) fails to work. That is caused because default `com.mongodb.async.client.MongoClientImpl.getDatabase` does not include the same codec registry as old `com.mongodb.MongoClient.getDatabase` that leads to runtime errors in serialization/deserialization.
Yes, it could be made customizable - in Mongo driver it is.